config/samples/resources/bigqueryconnectionconnection/bigqueryconnectionconnection_v1alpha1_bigqueryconnectionconnection.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: bigqueryconnection.cnrm.cloud.google.com/v1alpha1
+kind: BigQueryConnectionConnection
+metadata:
+  name: sample-cloudresource
+spec:
+  location: us-central1
+  projectRef:
+    external: ${PROJECT_ID?}
+  description: "BigQueryConnection CloudResource Connection"
+  cloudResource:
+    serviceAccountId: ""
```

Comment:
This is a tricky interface because `serviceAccountId` is actually output-only but got added under `spec`. Though it's okay to be used as a workaround in v1alpha1.

In order to populate its value under `status.observedState`, you should adopt the workaround in #2032.

Reply:
@maqiuyujoyce Thanks!
Do you want me to include the workaround in this PR, or will it be a separate one?

Reply:
Yes, you'll need to include the workaround in https://github.com/GoogleCloudPlatform/k8s-config-connector/commit/ee1d4b63427d69c57c7d3dbe8f4b2ed300a706fb to populate `status.observedState.serviceAccountId` properly during the reconciliation.

In addition, I suggest you run the VCR test and golden object comparison using the command to ensure this field is always populated properly:

```
E2E_KUBE_TARGET=envtest RUN_E2E=1 \
E2E_GCP_TARGET=vcr VCR_MODE=replay GOLDEN_OBJECT_CHECKS=1 \
go test -timeout 3600s -v ./tests/e2e -run TestAllInSeries/fixtures/^bigqueryconnectionconnection$
```

You'll want to generate/set up VCR cassette and the golden object first.
* Instructions for VCR testing: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/tests/README.VCRTesting.md
* To generate the golden object, you should add the environment variable `WRITE_GOLDEN_OUTPUT=1` during the test run. Some potentially helpful doc can be found here: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/mockgcp/README.md#capture-http-golden-logs

@gemmahou should be a good contact if you have any questions about the test.

Reply:
Here is the doc about establishing the test for observed state: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2128/files#diff-4ba365324b277bbbedf753f1d49fe7b02c3ee05cbccca1c72bd658f30ca14da8

Example can be found in the same PR.